bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java

```diff
@@ -639,4 +633,68 @@ static String getQueryForTimePartitionedTable(
         commaSeparatedFields,
         commaSeparatedFields);
   }
+
+  static String getQueryForTimePartitionedTable(
+      String destinationTableName,
+      String temporaryTableName,
+      StandardTableDefinition destinationDefinition,
+      TimePartitioning timePartitioning) {
+    TimePartitioning.Type partitionType = timePartitioning.getType();
+    String partitionField = timePartitioning.getField();
+    String extractedPartitioned = "timestamp_trunc(`%s`.`%s`, %s)";
+    String extractedPartitionedSource =
+        String.format(extractedPartitioned, "source", partitionField, partitionType.toString());
+    String extractedPartitionedTarget =
+        String.format(extractedPartitioned, "target", partitionField, partitionType.toString());
+    return getMergeQueryForPartitionedTable(
+        destinationTableName,
+        temporaryTableName,
+        destinationDefinition,
+        extractedPartitionedSource,
+        extractedPartitionedTarget);
+  }
+
+  static String getQueryForRangePartitionedTable(
+      String destinationTableName,
+      String temporaryTableName,
+      StandardTableDefinition destinationDefinition,
+      RangePartitioning rangePartitioning) {
+
+    long start = rangePartitioning.getRange().getStart();
+    long end = rangePartitioning.getRange().getEnd();
+    long interval = rangePartitioning.getRange().getInterval();
+
+    String partitionField = rangePartitioning.getField();
+    String extractedPartitioned =
+        "IF(%s.%s > %s, 0, RANGE_BUCKET(%s.%s, GENERATE_ARRAY(%s, %s, %s)))";
```

Comment:
Should this be >= ? 
The end is exclusive and not inclusive.
https://cloud.google.com/bigquery/docs/partitioned-tables#integer_range

Reply:
Good catch. Fixed!